### PR TITLE
Merge contract testing library

### PR DESCRIPTION
**Pull request by Bargsteen**
## Purpose

Merge the smart contract testing library into this repository.
This prepares us for migrating the old example tests to use the new testing library.

## Changes

- Merge in the library.
- Replace the contracts-common submodule with the rust-sdk (since it is needed by the testing lib and contains the contracts-common lib via the base submodule)
- Add a CI job for it that only runs when changes occur in the testing lib, the rust SDK or the rustfmt config file.
- Update the readme to the new reality.
- Checkout submodules recursively in the CI.
   - This is needed because the rust-sdk submodule has base as a submodule.  

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,3 @@
-[submodule "concordium-contracts-common"]
-	path = concordium-contracts-common
-	url = ../concordium-contracts-common.git
+[submodule "concordium-rust-sdk"]
+	path = concordium-rust-sdk
+	url = git@github.com:Concordium/concordium-rust-sdk.git
```
Comment (limemloh) [inline .gitmodules:3]: This should probably be kept, as a relative path
```suggestion
	url = ../concordium-rust-sdk.git
```

**Review by limemloh — APPROVED** [review]
LGTM

File: .gitmodules (hunk 1 above)
Comment (Bargsteen) [reply .gitmodules:3]: Ok, that's fine. I am not sure of the benefits though. Which do you see?

**Review by abizjak — APPROVED** [review]

File: .gitmodules (hunk 1 above)
Comment (abizjak) [reply .gitmodules:3]: It works regardless of whether you check out the repository via https or ssh.

Comment (Bargsteen) [reply .gitmodules:3]: Ah, neat.

**Pull request merged**